src/prost/wrapper_eraftpb.rs

```diff
@@ -163,40 +163,49 @@ impl SnapshotMetadata {
     pub fn take_conf_state(&mut self) -> ConfState {
         self.conf_state.take().unwrap_or_else(ConfState::default)
     }
-    pub fn has_pending_membership_change(&self) -> bool {
-        self.pending_membership_change.is_some()
+    pub fn clear_conf_state_index(&mut self) {
+        self.conf_state_index = 0
     }
-    pub fn clear_pending_membership_change(&mut self) {
-        self.pending_membership_change = ::std::option::Option::None
+    pub fn set_conf_state_index(&mut self, v: u64) {
+        self.conf_state_index = v;
     }
-    pub fn set_pending_membership_change(&mut self, v: ConfState) {
-        self.pending_membership_change = ::std::option::Option::Some(v);
+    pub fn get_conf_state_index(&self) -> u64 {
+        self.conf_state_index
     }
-    pub fn get_pending_membership_change(&self) -> &ConfState {
-        match self.pending_membership_change.as_ref() {
+    pub fn has_next_conf_state(&self) -> bool {
```

Comment:
If `get_next_conf_state`  returned an option then this would be unnecessary? 🤔 It seems a bit awkward. It's not a problem though, but I think it's confusing.

Reply:
It's generated by prost. I guess it follows CPP's style. I also think it's weird and it's easy to make mistakes. Could we change the codegen to return an `Option`? @ice1000

Reply:
Correction: it's not generated by prost but by our wrapper generator. Prost structs' fields are public, there's no need of getters/setters/hases. These methods are generated to ease the migration from rust-protobuf to prost.

Reply:
There must be a similar method that rust-protobuf will generate, too.
This can be an issue to github.com/nrc/kvproto-build

Reply:
Let's leave it for now.